### PR TITLE
EDU-458 Allow login with email address or username

### PR DESCRIPTION
**Pull request by ahelmberger**
Closes https://educandu.atlassian.net/browse/EDU-458

### CODE REVIEW

File: src/stores/user-store.js [hunk 1]
```diff
@@ -11,8 +11,25 @@ class UserStore {
...
+    }).toArray();
+
+    switch (matches.length) {
+      case 0:
+        return null;
+      case 1:
+        return matches[0];
+      default:
+        // Always prefer matching emails over usernames:
+        return matches.find(match => match.email === emailOrUsername);
```
Comment (adrianaluchian) [inline src/stores/user-store.js:31]: Micro thing, but 
```
const preferredMatch = matches.find(match => match.email === emailOrUsername);
return preferredMatch;
```
would speak for itself without a comment that risks getting out of sync with code :bowtie:

File: src/stores/user-store.spec.js [hunk 2]
```diff
@@ -25,28 +25,37 @@ describe('user-store', () => {
...
+        result = await sut.findUserByLogin({ provider: user.provider, emailOrUsername: user.email });
+      });
+      it('should return the user', () => {
+        expect(result).toEqual(user);
+      });
+    });
+
+    describe('when provider matches and emailOrUsername matches the username', () => {
+      beforeEach(async () => {
+        result = await sut.findUserByLogin({ provider: user.provider, emailOrUsername: user.username });
```
Comment (adrianaluchian) [inline src/stores/user-store.spec.js:58]: Would be good to also add a test for both username and email matching.

File: src/components/login-form.js [hunk 3]
```diff
@@ -1,8 +1,7 @@
 import PropTypes from 'prop-types';
```
Comment (adrianaluchian) [inline src/components/login-form.js:1]: Both the code here and the UI look alot better now 👍 
Maybe we could make a ticket to update registration in the same direction.

File: src/stores/user-store.js (hunk 1 above)
Comment (ahelmberger) [reply src/stores/user-store.js:31]: Changed it, my commentophobic friend :laughing:

Comment (adrianaluchian) [reply src/stores/user-store.js:31]: 😄 thanks! I would say rather a deep dislike than a phobia, I rather have phobia of code that needs comments 😨

File: src/stores/user-store.spec.js (hunk 2 above)
Comment (ahelmberger) [reply src/stores/user-store.spec.js:58]: Yep, good idea, added a test for this case.

File: src/stores/user-store.js (hunk 1 above)
Comment (ahelmberger) [reply src/stores/user-store.js:31]: :grin:

File: src/components/login-form.js (hunk 3 above)
Comment (ahelmberger) [reply src/components/login-form.js:1]: Yes, I created a ticket: https://educandu.atlassian.net/browse/EDU-462

**Review by adrianaluchian — APPROVED** [review]

**Pull request merged**